fili-core/src/test/groovy/com/yahoo/bard/webservice/logging/blocks/PrefaceSpec.groovy

```diff
@@ -42,6 +49,65 @@ class PrefaceSpec extends Specification {
         }
 
         expect: "Making a Preface with it sets the user to the user name"
-        new Preface(mockRequest).user == knownUserName
+        def preface = new Preface(mockRequest)
+        preface.getUser() == knownUserName
+        MAPPER.valueToTree(preface).toString() == """{"uri":"http://example.com","method":"","headers":{"Cookie":["Cookies not present in header"]},"user":"${knownUserName}"}"""
+    }
+
+    def "User set to username value at serialization time"() {
+        given: "A request with a userPrincipal"
+        def securityContext = new MockSecurityContext(null)
+        mockRequest.getSecurityContext() >> securityContext
+        def preface = new Preface(mockRequest)
+
+        expect: "Use default username value"
+        preface.getUser() == Preface.NO_USER_PRINCIPAL_USER_NAME
+
+        when: "Set the username value"
+        String knownUserName = "Known User Name"
+        def userPrincipal = new MockUserPrincipal(knownUserName)
```

Comment:
If this is only gonna use once, a anonymous class is good enough. Maybe something like this:
```
def userPrincipal = new UserPrincipal() {
            private final String name = knownUserName

            @Override
            String getName() {
                return name
            }

            @Override
            boolean implies(Subject subject) { return false }
        }
```
Same for `MockSecurityContext`.
Also the name `MockUserPrincipal` is easily confused with a object created with Mock.